synapseclient/client.py

```diff
@@ -1274,253 +1275,266 @@ def store(
             test_entity = syn.store(test_entity, activity=activity)
 
         """
-        # SYNPY-1031: activity must be Activity object or code will fail later
-        if activity:
-            if not isinstance(activity, synapseclient.Activity):
-                raise ValueError("activity should be synapseclient.Activity object")
-        # _before_store hook
-        # give objects a chance to do something before being stored
-        if hasattr(obj, "_before_synapse_store"):
-            obj._before_synapse_store(self)
-
-        # _synapse_store hook
-        # for objects that know how to store themselves
-        if hasattr(obj, "_synapse_store"):
-            return obj._synapse_store(self)
-
-        # Handle all non-Entity objects
-        if not (isinstance(obj, Entity) or type(obj) == dict):
-            if isinstance(obj, Wiki):
-                return self._storeWiki(obj, createOrUpdate)
-
-            if "id" in obj:  # If ID is present, update
-                trace.get_current_span().set_attributes({"synapse.id": obj["id"]})
-                return type(obj)(**self.restPUT(obj.putURI(), obj.json()))
-
-            try:  # If no ID is present, attempt to POST the object
-                trace.get_current_span().set_attributes({"synapse.id": ""})
-                return type(obj)(**self.restPOST(obj.postURI(), obj.json()))
-
-            except SynapseHTTPError as err:
-                # If already present and we want to update attempt to get the object content
-                if createOrUpdate and err.response.status_code == 409:
-                    newObj = self.restGET(obj.getByNameURI(obj.name))
-                    newObj.update(obj)
-                    obj = type(obj)(**newObj)
+        with tracer.start_as_current_span(
```

Comment:
Adding this `with` is the only real change here - The sync functions multi-thread at this point and OTEL context was not propagating

Reply:
Do you mind adding the explanation of opentelemetry_context parameter in the function docstring?